store/postgres/src/mapping_sql.rs

```diff
@@ -0,0 +1,1107 @@
+                let s = if starts_with {
+                    format!("{}%", s)
+                } else {
+                    format!("%{}", s)
+                };
+                out.push_bind_param::<Text, _>(&s)?
+            }
+            Value::Bool(_)
+            | Value::BigInt(_)
+            | Value::Bytes(_)
+            | Value::BigDecimal(_)
+            | Value::Int(_)
+            | Value::List(_)
+            | Value::Null => {
+                return Err(UnsupportedFilter {
+                    filter: op.to_owned(),
+                    value: value.clone(),
+                }
+                .into());
+            }
+        }
+        Ok(())
+    }
+}
+
+impl<'a> QueryFragment<Pg> for QueryFilter<'a> {
+    fn walk_ast(&self, mut out: AstPass<Pg>) -> QueryResult<()> {
+        out.unsafe_to_cache_prepared();
+
+        use EntityFilter::*;
+        match &self.filter {
+            And(filters) => self.binary_op(filters, " and ", out)?,
+            Or(filters) => self.binary_op(filters, " or ", out)?,
+
+            Contains(attr, value) => self.contains(attr, value, false, out)?,
+            NotContains(attr, value) => self.contains(attr, value, true, out)?,
+
+            Equal(attr, value) => self.equals(attr, value, false, out)?,
+            Not(attr, value) => self.equals(attr, value, true, out)?,
+
+            GreaterThan(attr, value) => self.compare(attr, value, " > ", out)?,
+            LessThan(attr, value) => self.compare(attr, value, " < ", out)?,
+            GreaterOrEqual(attr, value) => self.compare(attr, value, " >= ", out)?,
+            LessOrEqual(attr, value) => self.compare(attr, value, " <= ", out)?,
+
+            In(attr, values) => self.in_array(attr, values, false, out)?,
+            NotIn(attr, values) => self.in_array(attr, values, true, out)?,
+
+            StartsWith(attr, value) => {
+                self.starts_or_ends_with(attr, value, " like ", true, out)?
+            }
+            NotStartsWith(attr, value) => {
+                self.starts_or_ends_with(attr, value, " not like ", true, out)?
+            }
+            EndsWith(attr, value) => self.starts_or_ends_with(attr, value, " like ", false, out)?,
+            NotEndsWith(attr, value) => {
+                self.starts_or_ends_with(attr, value, " not like ", false, out)?
+            }
+        }
+        Ok(())
+    }
+}
+
+#[derive(Debug, Clone, Constructor)]
+pub struct FindQuery<'a> {
+    mapping: &'a Mapping,
+    entity: &'a str,
+    id: &'a str,
+    block: BlockNumber,
+}
+
+impl<'a> FindQuery<'a> {
+    fn object_query(&self, table: &'a Table, mut out: AstPass<Pg>) -> QueryResult<()> {
+        // Generate
+        //    select '..' as entity, to_jsonb(e.*) as data from schema.table e where id = $1
+        out.push_sql("select ");
+        out.push_bind_param::<Text, _>(&self.entity)?;
+        out.push_sql(" as entity, to_jsonb(e.*) as data\n");
+        out.push_sql("  from ");
```

Comment:
Now I remember why that's intentional: it makes it so that the query is nicely aligned when we are done, i.e. it generates something like
```
select thing as entity, ...
  from some_table
```

Makes it a little nicer when looking at the Postgres log when these things get logged.

Reply:
Ah I see now, could you update the comment above to reflect the multiline format that's being used?